include/pybind11/numpy.h

```diff
@@ -220,6 +219,16 @@ inline bool check_flags(const void* ptr, int flag) {
     return (flag == (array_proxy(ptr)->flags & flag));
 }
 
+template <typename T> struct is_std_array : std::false_type { };
+template <typename T, size_t N> struct is_std_array<std::array<T, N>> : std::true_type { };
+template <typename T> struct is_complex : std::false_type { };
+template <typename T> struct is_complex<std::complex<T>> : std::true_type { };
+
+template <typename T> using is_pod_struct = all_of<
+    std::is_pod<T>, // offsetof only works correctly for POD types
+    is_none_of<T, std::is_reference, std::is_array, is_std_array, std::is_arithmetic, is_complex, std::is_enum>
```

Comment:
Because it was before.  (I didn't change the logic here, just cleaned it up a bit with convenience meta-templates).  I seem to recall some discussion about coming back to handle that eventually (cc @aldanor?) at some point post 2.0.